zipline/utils/events.py

```diff
@@ -594,13 +605,14 @@ def get_nth_to_last_trading_day_of_month(self, dt, env):
             # We already computed the last day for this month.
             return self.day
 
-        if not self.td_delta:
-            self.day = self.get_last_trading_day_of_month(dt, env)
-        else:
-            self.day = env.add_trading_days(
-                self.td_delta,
-                self.get_last_trading_day_of_month(dt, env),
-            ).date()
+        self.day = self.get_last_trading_day_of_month(dt, env)
+        if self.td_delta and self.day:
+            try:
+                self.day = env.add_trading_days(self.td_delta, self.day)
+            except NoFurtherDataError:
+                self.day = None
+            if self.day:
```

Comment:
`add_trading_days` could potentially return `None` as well, so we need to account for that